Loop/Window Management/Window/WindowUtility+FocusNavigation.swift

```diff
@@ -0,0 +1,120 @@
+//
+//  WindowUtility+FocusNavigation.swift
+//  Loop
+//
+//  Created by cipher-shad0w on 2025-10-30.
+//
+
+import AppKit
+import OSLog
+import SwiftUI
+
+extension WindowUtility {
+    private static let focusLogger = Logger(category: "WindowUtility+FocusNavigation")
+
+    /// Finds the next window to focus in the specified direction.
+    /// - Parameters:
+    ///   - currentWindow: The currently focused window to navigate from, or nil to navigate from screen center
+    ///   - direction: The direction to search for the next window (focusUp, focusDown, focusLeft, focusRight)
+    /// - Returns: The next window in the specified direction, or `nil` if no suitable window is found
+    static func nextWindow(from currentWindow: Window?, in direction: WindowDirection) -> Window? {
```

Comment:
I feel like "next" is contradictory with the term "direction" as they mean two different things in screen switching. Also, since the direction cannot be *any* direction, it would make sense to check if `direction.focusEdge` exists *before* calling this function:

```suggestion
    static func directionalWindow(from currentWindow: Window?, in edge: Edge) -> Window? {
```